src/preact/component/component.type.js

```diff
@@ -43,3 +43,8 @@ var WrapperComponentProps;
  * }}
  */
 var ContainWrapperComponentProps;
+
+/**
+ * @typedef {function(!JsonObject):(?string|!Object|!PreactDef.Renderable)}
```

Comment:
Why include `!Object`?

Reply:
`!Object` is the `{__html: ...}` shape. I could potentially add it in our standard set of extern types and use it here explicitly?

Reply:
I went ahead and defined an extern for the HTML response.